proxy/http2/HTTP2.cc

```diff
@@ -406,88 +406,82 @@ http2_parse_window_update(IOVec iov, uint32_t &size)
 ParseResult
 http2_convert_header_from_2_to_1_1(HTTPHdr *headers)
 {
-  MIMEField *field;
-
   ink_assert(http_hdr_type_get(headers->m_http) != HTTP_TYPE_UNKNOWN);
 
+  // HTTP Version
+  headers->version_set(HTTPVersion(1, 1));
+
   if (http_hdr_type_get(headers->m_http) == HTTP_TYPE_REQUEST) {
-    const char *scheme, *authority, *path;
-    int scheme_len, authority_len, path_len;
+    // :scheme
+    if (MIMEField *field = headers->field_find(HTTP2_VALUE_SCHEME, HTTP2_LEN_SCHEME); field != nullptr && field->value_is_valid()) {
+      int scheme_len;
+      const char *scheme = field->value_get(&scheme_len);
 
-    // Get values of :scheme, :authority and :path to assemble requested URL
-    if ((field = headers->field_find(HTTP2_VALUE_SCHEME, HTTP2_LEN_SCHEME)) != nullptr && field->value_is_valid()) {
-      scheme = field->value_get(&scheme_len);
+      int scheme_wks_idx = hdrtoken_tokenize(scheme, scheme_len);
+      url_scheme_set(headers->m_heap, headers->m_http->u.req.m_url_impl, scheme, scheme_wks_idx, scheme_len, true);
+
+      headers->field_delete(field);
     } else {
       return PARSE_RESULT_ERROR;
     }
 
-    if ((field = headers->field_find(HTTP2_VALUE_AUTHORITY, HTTP2_LEN_AUTHORITY)) != nullptr && field->value_is_valid()) {
-      authority = field->value_get(&authority_len);
+    // :authority
+    if (MIMEField *field = headers->field_find(HTTP2_VALUE_AUTHORITY, HTTP2_LEN_AUTHORITY);
+        field != nullptr && field->value_is_valid()) {
+      int authority_len;
+      const char *authority = field->value_get(&authority_len);
+
+      url_host_set(headers->m_heap, headers->m_http->u.req.m_url_impl, authority, authority_len, true);
+
+      headers->field_delete(field);
     } else {
       return PARSE_RESULT_ERROR;
     }
 
-    if ((field = headers->field_find(HTTP2_VALUE_PATH, HTTP2_LEN_PATH)) != nullptr && field->value_is_valid()) {
-      path = field->value_get(&path_len);
+    // :path
+    if (MIMEField *field = headers->field_find(HTTP2_VALUE_PATH, HTTP2_LEN_PATH); field != nullptr && field->value_is_valid()) {
+      int path_len;
+      const char *path = field->value_get(&path_len);
+
+      // cut first '/' if there, because `url_print()` add '/' before printing path
+      if (path_len >= 1 && path[0] == '/') {
```

Comment:
Is this the same as what `url_parse` does? What if there are multiple slashes?

Reply:
I had the same question for multiple slashes and tried. If my experiment is correct, the only first `/` is removed.

Reply:
Ok, I'm fine fine with this as long as the behavior is the same.